crates/sui-faucet/src/metrics_layer.rs

```diff
@@ -69,12 +69,14 @@ where
         let future = Box::pin(async move {
             metrics.total_requests_received.inc();
             metrics.current_requests_in_flight.inc();
+            let _metrics_guard = scopeguard::guard(metrics.clone(), |metrics| {
+                metrics.current_requests_in_flight.dec();
+            });
 
             let timer = metrics.process_latency.start_timer();
             let resp = inner.oneshot(req).await;
 
             let elapsed = timer.stop_and_record();
```

Comment:
Similar question, should we be counting the number of requests that end this way? (Dropped without explicit success or failure).  Not clear to me the answer is yes because you can figure the number out using the existing metrics, but it would be clearer to count them explicitly.

Reply:
@amnn it turns out the elapsed timer handles out of scope correctly: https://docs.rs/prometheus/0.3.2/src/prometheus/histogram.rs.html#338-342

Reply:
> should we be counting the number of requests that end this way?  

Good point, I will handle this in a separate PR

Reply:
Nice, thanks for checking!